src/rosdiscover/cli.py

```diff
@@ -58,12 +59,29 @@ def generate_acme(args):
     if args.acme is None:
         print(acme)
         if args.check:
-            acme_gen.check_acme(acme)
+            acme_gen.check_acme()
     else:
         if args.check:
             acme_gen.check_acme()
 
 
+def _observe(args) -> SystemSummary:
+    config = Config.from_yaml_string(args.config)
+    with Observer.for_container(args.container, config) as obs:
+        summary = obs.observe_and_summarise()
+    return summary
+
+
+def observe(args) -> None:
+    summary = _observe(args)
```

Comment:
I'm not sure why `_observe` is a separate method. It seems like the first three lines of `_observe` could be dropped inlined at this location.

Reply:
Am preparing for when we also want it to output Acme in addition to yaml. (There will be a smaller PR after this one to change the CLI to be more uniform about the Acme).